include/alpaka/core/BoostPredef.hpp

```diff
@@ -40,10 +39,12 @@
 #    endif
 #endif
 
-// hip compiler detection
+// HIP compiler detection
 #if !defined(BOOST_COMP_HIP)
-#    if defined(__HIP__)
-#        define BOOST_COMP_HIP BOOST_VERSION_NUMBER_AVAILABLE
+#    if defined(__HIPCC__) // We are using a HIP-capable compiler; could be hcc, hip-clang or nvcc
+#        include <hip/hip_version.h>
+// HIP doesn't give us a patch level for the last entry, just a gitdate
+#        define BOOST_COMP_HIP BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, 0)
```

Comment:
Essentially, `BOOST_COMP_HIP` and `BOOST_LANG_HIP` do the same thing since there is no special HIP language we could test for. If we use a HIP-capable compiler we assume everything it touches is a "HIP source file".